plugins/cilium-cni/chaining/generic-veth/generic-veth.go

```diff
@@ -107,6 +107,15 @@ func (f *GenericVethChainer) Add(ctx context.Context, pluginCtx chainingapi.Plug
 			}
 
 			vethIP = addrs[0].IPNet.IP.String()
+
+			addrsv6, err := netlink.AddrList(link, netlink.FAMILY_V6)
+
+			if err != nil {
+				vethIPv6 = ""
```

Comment:
I don't think this is needed. You can simply write:
```go
if err == nil {
    vethIPv6 = addrsv6[0].IPNet.IP.String()
}
```

That said, if there is an error, maybe we want to log it somehow?

Reply:
hmm, error could be that there's simply no ipv6 (not enabled), not sure if we wanna log that?

Reply:
If there isn't an IPv6 address, I would just expect addrsv6 to be an empty slice. We should also check for that by the way, or we may end up with a panic :grimacing: 

Errors seem to happen if the initial netlink message fails or if the responses can't be parsed: https://github.com/vishvananda/netlink/blob/f049be6f391489d3f374498fe0c8df8449258372/addr_linux.go#L180.

Reply:
ah you are right! Added a check and also added a log entry on failure.